app/controllers/categories_controller.rb

```diff
@@ -286,9 +286,13 @@ def category_params
       end
 
       if SiteSetting.tagging_enabled
-        params[:allowed_tags] ||= []
-        params[:allowed_tag_groups] ||= []
-        params[:required_tag_group_name] ||= ''
+        params[:allowed_tags] = params[:allowed_tags].presence || [] if params[:allowed_tags]
```

Comment:
Will the `|| []` ever be called? 

```
[3] pry(main)> nil.presence || [] if nil
=> nil
[4] pry(main)> true.presence || [] if true
=> true
[5] pry(main)> false.presence || [] if false
=> nil
```

If `params[:allowed_tags]` is `nil`, it'll not pass the conditional so `params[:allowed_tags].presence` in this case can never be `false` or `nil`

Reply:
Yes, because not all true values are also present.

When there are no more allowed_tags, params[:allowed_tags] will be equal to "". The if makes sure that allowed_tags will be updated and and the .presence and || make sure that the blank string is converted to an empty array. It also works well when params[:allowed_tags] is equal to an array of strings.

Reply:
ahhh I didn't know `presence` actually returns `nil` for blank strings. All good here then :+1: